src/map.cpp

```diff
@@ -610,7 +612,9 @@ void ServerMap::transformLiquids(std::map<v3s16, MapBlock*> &modified_blocks,
 			v3s16 npos = p0 + liquid_6dirs[i];
 			NodeNeighbor nb(getNode(npos), nt, npos);
 			const ContentFeatures &cfnb = m_nodedef->get(nb.n);
-			switch (m_nodedef->get(nb.n.getContent()).liquid_type) {
+			if (nt == NEIGHBOR_UPPER && cfnb.floats)
```

Comment:
This should be run for all falling nodes which could be generated by the mapgen or manually with worldedit or other mods.

Reply:
In my view that would break compatibility.

Reply:
Compatibility with what?

Reply:
Say you explore some caves with dynamic_liquid enabled. Due to liquid movement, suddenly the suspended sand in proximity to the water would fall without you having to punch it.

Reply:
I think this would be a bad idea too, but in any case it's for another PR